### PR TITLE
New plugin: AutoThemeSwitcher

### PR DESCRIPTION
**Pull request by maddie480**
So, here is a plugin I made for myself a bit ago, for switching between a "light" and a "dark" theme based on the time of day: 

attachment

This matches up with a plugin request that was made 2 weeks ago (https://github.com/Vencord/plugin-requests/issues/772), but still, feel free to tell me if this would be better as a third-party plugin :sweat_smile:

### CODE REVIEW

**Review by Scyye — COMMENTED** [review]
1. this seems like *a lot* of files
2. I'd rename the files, casingLikeThis, rather than like-this

File: src/plugins/autoThemeSwitcher/plugin-settings.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        },
+        lightTheme: {
+            description: "Light Theme",
+            type: OptionType.SELECT,
+            options: proxyLazy(() => themeLister.getSelectOptions(ToggledTheme.Light)),
+            onChange
+        },
+        lightThemeURLs: {
+            description: "Light Theme CSS URLs (1 per line)",
+            type: OptionType.STRING_MULTILINE,
```
[the remark this answers is not in the document]
Comment (maddie480) [reply src/plugins/autoThemeSwitcher/plugin-settings.ts:36]: I used a text field with values separated by ; at first, but I figured a text area that worked the same way as the actual Theme Links option would be nicer 😅 and that maybe other plugins could want to have multiline text fields too
But I can definitely go back to using ;

[the remark this answers is not in the document]
Comment (Scyye) [reply src/plugins/autoThemeSwitcher/plugin-settings.ts:36]: thats fair, i can see other plugins using it. just asking as to the reasoning.

**Comment by sadan4** [comment]
this plugin seems better done via the native discord feature of syncing with your computer theme. 
image

**Comment by maddie480** [comment]
> this plugin seems better done via the native discord feature of syncing with your computer theme. 
> image
> 

The plugin lets you select other themes besides Light and Dark, unsure how niche this is though 🤔

And I never got the system theme option to work on my Linux Mint 😅 no clue why

**Comment by maddie480** [comment]
It's the second time I'm seeing comments about the new option type (including core-development on the Discord), should I switch to a comma-separated list? I just noticed that it's what MessageLogger uses 🤔

**Comment by maddie480** [comment]
I'll probably come back with another PR to integrate "different light and dark themes" in theme settings directly, to make it less confusing. Rather than following its own schedule, it could just rely on the active Discord theme, and "sync with system" would give auto theme switching for free.

... But before I can personally use that, there's still the mystery of "why does 'sync with system' always end up setting the theme to Light on my computer" left to solve 🤔 It's either a Chromium bug or a skill issue from me, still not sure. In the meantime, I turned this PR into a third party plugin: https://github.com/maddie480/Vencord-AutoThemeSwitcher

**Pull request closed without merging**